### PR TITLE
AirbyteLib: More robust error handling, installation improvements

### PR DESCRIPTION
**Pull request by aaronsteers**
This does a few things:

- Remove `-e` (editable) flag from install. Users can still get an editable install by using `-e ./path/to/my-connector` in pip_url. (This was done because -e seemed to fail when used with installations that were not from local paths.)
- Improve the stderr output from pip when printing installation exceptions.
- Clean up a failed installation so users can try again.
- Move certain checks out of the constructors (or make optional), so we can still return an object for the user to use in debugging.
- Solve some issues where running tests in parallel would bleed into each other tests and/or create race conditions.

### CODE REVIEW

**Review by aaronsteers — COMMENTED** [review]
I was unable to debug a new notebook I was creating for GitHub analytics. This PR addresses some of those challenges I ran into.

File: airbyte-integrations/connectors/source-github/setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     author_email="contact@airbyte.io",
     packages=find_packages(),
     install_requires=MAIN_REQUIREMENTS,
-    package_data={"": ["*.json", "schemas/*.json", "schemas/shared/*.json"]},
+    package_data={"": ["**/*.json"]},
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-github/setup.py:19]: revert

```suggestion
    package_data={"": ["*.json", "schemas/*.json", "schemas/shared/*.json"]},
```

File: airbyte-lib/tests/integration_tests/test_integration.py [hunk 2]
```diff
@@ -26,24 +27,32 @@
...
+
+
+@pytest.fixture(scope="package", autouse=True)
 def prepare_test_env():
     """
     Prepare test environment. This will pre-install the test source from the fixtures array and set the environment variable to use the local json file as registry.
     """
+    venv_dir = f".venv-source-test"
     if os.path.exists(".venv-source-test"):
         shutil.rmtree(".venv-source-test")
```
Comment (aaronsteers) [inline airbyte-lib/tests/integration_tests/test_integration.py:40]:
```suggestion
    venv_dir = ".venv-source-test"
    if os.path.exists(venv_dir):
        shutil.rmtree(venv_dir)
```

**Review by flash1293 — APPROVED** [review]
Thanks for these changes, I agree with most of it, left some nit and one question.

File: airbyte-lib/airbyte_lib/registry.py [hunk 3]
```diff
@@ -12,47 +14,81 @@
 from airbyte_lib.version import get_version
 
 
+__cache: dict[str, ConnectorMetadata] | None = None
+_cache_lock = threading.Lock()
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte-lib/airbyte_lib/registry.py:18]: Good catch. I had it in an earlier version when debugging a race condition, then refactored so it wasn't needed.

File: airbyte-lib/airbyte_lib/source.py [hunk 4]
```diff
@@ -102,8 +109,22 @@ def set_streams(self, streams: list[str]) -> None:
                 )
         self._selected_stream_names = streams
 
-    def set_config(self, config: dict[str, Any]) -> None:
-        self._validate_config(config)
+    def set_config(
+        self,
+        config: dict[str, Any],
+        *,
+        validate: bool = False,
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte-lib/airbyte_lib/source.py:116]: Discussed in DM.

File: airbyte-lib/airbyte_lib/exceptions.py [hunk 5]
```diff
@@ -185,7 +193,7 @@ class AirbyteConnectorError(AirbyteError):
 
 
 class AirbyteConnectorNotFoundError(AirbyteConnectorError):
-    """Connector not found."""
+    """Connector name not found in registry."""
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte-lib/airbyte_lib/exceptions.py]: Good catch! This was a mistake. I've renamed the error to `AirbyteConnectorExecutableNotFoundError` so it is more explicit.

**Pull request merged**